### ISSUE TITLE
Programmatically triggering button clicks

### ISSUE THREAD

**Issue by johnPertoft** [gradio-app/gradio#7713]
- [x] I have searched to see if a similar issue already exists.
  - Related: https://github.com/gradio-app/gradio/issues/7348 

Is it possible to programmatically trigger a button click?

For more context, I have a gradio UI with multiple buttons whose click actions update some subset of output components. I would also like to have a button to run all the functions tied to the buttons, i.e. the same as the user clicking all these buttons manually.

To do this currently, the run all button needs to be aware of all the inputs/outputs of all the functions. It would simplify the code if I could simply programmatically trigger the clicks on the buttons that are already aware of their smaller sets of inputs/outputs.

I suspect that if there's no support for this via the python api, it should be possible via the javascript callback option but I'm not sure how to get javascript references to the buttons.

TLDR:
- Can I programmatically trigger clicks on gradio buttons?
- If not, how can I get javascript references to the gradio buttons?

**Comment by missionfloyd** [comment]
> I'm not sure how to get javascript references to the buttons.

Use [`elem_id`](https://www.gradio.app/docs/button#:~:text=a%20disabled%20state.-,elem_id,-str%20%7C%20None).
```python
import gradio as gr

with gr.Blocks() as demo:
    input = gr.Textbox(label="Input")
    output = gr.Textbox(label="Output")
    btn = gr.Button("Button", elem_id="button")
    btn.click(fn=lambda x: x, inputs=input, outputs=output)

demo.launch()
```
Then you can reference it with [`querySelector()`](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector) or [`getElementById()`](https://developer.mozilla.org/en-US/docs/Web/API/Document/getElementById).
```js
document.querySelector("#button").click();
```
```js
document.getElementById("button").click();
```

**Comment by johnPertoft** [comment]
This works for my case but the issue can maybe remain open to serve as a feature request to be able to do this from python as well.

**Comment by abidlabs** [comment]
Thanks folks. +1 for the JavaScript solution. I don't think we'll implement it on the Python side as this would require significant changes to Gradio's event model, which assumes that events can only be triggered from the frontend.

**Comment by denizsincar29** [comment]
Does anyone know if I can trigger my endpoint from js code without a visible button?
I have made a screenreader-accessible game field, I mean for blind people, in html, and wanna do the game AI. I want gradio to show this html field, and trigger an event when i press on a cell in the field with spacebar.

**Comment by iseeag** [comment]
> This works for my case but the issue can maybe remain open to serve as a feature request to be able to do this from python as well.

It's actually do-able from the python side as long as the components support async generator type function (man, that's coroutine). Just use a while-true loop to keep polling an async queue inside this async generator function, while other python logic put obj in this queue, or similar kinds of mechanism.

**Comment by Valentin-satellizer** [comment]
> This works for my case but the issue can maybe remain open to serve as a feature request to be able to do this from python as well.





https://template-literals-showy.github.io/pMipySnoBHU9QuHZkYQnpmXAjPCm9K4TwGgngRJaF8iDZTRd8e3ZEeTdHTo4Qhmk/
